### PR TITLE
Integrating Travis CI: Debate!

### LINKED ISSUE #161: Continuous Integration
It's time we looked into bring a continuous integration tool into our pipeline. [GitHub](https://github.com/integrations/feature/continuous-integration) provides some good integrations with C.I. tools.  
Let's discuss.

### LINKED ISSUE #163: Test pathing
Various tests use absolute pathing that force that tests to have to be run from a specific spot. 
Example:
```python
sys.path.insert(0, os.path.abspath('../../../..'))
``` 
A better solution needs to be figured out.

### PR DESCRIPTION
**Pull request by isaac-gs**
@asclines @RyanMcBerg @ZakeryFyke 

Any opinions?

Includes and closes #161 and closes #163 ,
- Standard test data
- Travis CI integration files
- setupy.py and necessary files
- requirements.txt (things needed to run framework)
- necessary changes to tests

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -0,0 +1,8 @@
+MENGEL is a automated machine learning framework where the long-term goal
+is to make it distributable between multiple machines.
+
+All documentation necessary exists on our GitHub.
+
+To contribute, read our CONTRIBUTING.md file.
+
+* Add to this to make it more in line with example: https://github.com/django/django/blob/master/README.rst
```
Comment (asclines) [inline README.rst]: Whats this line for?

File: path_handler.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
+import os
```
Comment (asclines) [inline path_handler.py:1]: Is this file only used for testing? If so can we move it into the tests folder?

File: test_file_downloader.py [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import urllib
```
Comment (asclines) [inline test_file_downloader.py]: Is this file only used for testing? If so can we move it into the tests folder?

File: tests/unittest_data/demo.csv [hunk 4]
```diff
@@ -0,0 +1,10 @@
+demo,Name,gender,demo two,demo num,demo_cat,target
```
Comment (asclines) [inline tests/unittest_data/demo.csv]: `test_data` might be a better name for this folder as we will have system tests to that use this data?

File: path_handler.py (hunk 2 above)
Comment (asclines) [reply path_handler.py:1]: Update: Okay I see that is not just used for testing, but this shouldn't be used for the source. It technically brings a coupling of test code and source code. Also, the pathing issue isn't a problem for the framework.

File: setup.py [hunk 5]
```diff
@@ -1,11 +1,17 @@
...
-
-def get_root_path():
-    return os.path.dirname(os.path.abspath(inspect.getfile(inspect.currentframe())))
+# Further documentation on this topic
+# https://packaging.python.org/distributing/
+setup(name="MENGEL",
+      version="0.1",
+      description="Automated machine learning framework in Python.",
+      author="Code_Space",
+      author_email="alexander.clines@ttu.edu",
```
Comment (asclines) [inline setup.py:9]: lol what

**Review by asclines — CHANGES_REQUESTED** [review]
Overall looks good. But I really, really do not like path handler. There has to be a better way.. right?

File: path_handler.py (hunk 2 above)
Comment (isaac-gs) [reply path_handler.py:1]: What's your suggestion? I realize we probably shouldn't have it. But it makes pathing much easier. We could define global variables. Haha

File: README.rst (hunk 1 above)
Comment (isaac-gs) [reply README.rst]: What line?

File: path_handler.py (hunk 2 above)
Comment (asclines) [reply path_handler.py:1]: Global variables for what? I don't remember having pathing issues with the framework?

**Review by asclines — APPROVED** [review]

**Comment by asclines** [comment]
Approved, but please make sure Travis CI checks are in place before merging.

We can worry about the pathing problem in a future issue/PR, this PR should be focused on Travis CI

File: .travis.yml [hunk 6]
```diff
@@ -6,7 +6,7 @@ install:
 - "pip install --upgrade pip"
 - "pip install -r requirements.txt"
 before_script:
-- python /tests/unittest_data/test_file_downloader.py
+- python tests/unittest_data/test_file_downloader.py
```
Comment (asclines) [inline .travis.yml:9]: lol that was it?

**Pull request merged**